test/data.spec.js

```diff
@@ -105,3 +105,22 @@ test('Proxies are not nested and duplicated when manipulating an array', async (
     document.querySelector('button').click()
     await wait(() => { expect(document.querySelector('span').innerText).toEqual('bar') })
 })
+
+test('component does not refresh multiple times when data changes', async () => {
+    window.renderCount = 0
+
+    document.body.innerHTML = `
+        <div x-data="{ items: [], test() {return ++renderCount}  }">
```

Comment:
Ah this test again, very nice 💪💪

Reply:
Haha. I just copied Ahmed's test. All credits to him. ;)

Reply:
Good job guys, I will close #271